### PR TITLE
style: fix sass version >= 1.77.7, this use of declarations after nes…

### LINKED ISSUE #2366: [BUG] Sass version >= 1.77.7,  the use of declarations after nested rules is currently deprecated in Sass
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Which Component

Datepicker

### Semi Version

2.62.1

### Current Behavior

img_v3_02d2_d965bfbb-fc5b-435c-ba5c-22feff42a9hu

### Expected Behavior

sass 不报 waring，正常使用

### Steps To Reproduce

_No response_

### ReproducibleCode

_No response_

### Environment

```markdown
- OS:
- browser:
```

### Anything else?

sass新版本变更
https://sass-lang.com/documentation/breaking-changes/mixed-decls/

### PR DESCRIPTION
**Pull request by YyumeiZhang**
…ted rules deprecated

[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [ ] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [ ] Bugfix
 - [ ] Feature
 - [x] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

Fixes #2366
https://sass-lang.com/documentation/breaking-changes/mixed-decls/
原来的方式：
image
image

### Changelog
🇨🇳 Chinese
- Style:  修复 Sass 版本大于等于 1.77.7，不推荐使用嵌套规则后的声明问题 #2366 

---

🇺🇸 English
- Style: fixed the problem that the use of declarations after nested rules is currently deprecated in Sass(version >= 1.77.7) #2366 

### Checklist
- [ ] Test or no need
- [ ] Document or no need
- [ ] Changelog or no need

### Other
- [ ] Skip Changelog

### CODE REVIEW

File: packages/semi-foundation/navigation/navigation.scss [hunk 1]
```diff
@@ -438,7 +438,9 @@ $module: #{$prefix}-navigation;
                 &-selected {
                     font-weight: $font-navigation_sub_selected-fontWeight;
                     @include item-selected;
-                    background-color: $color-navigation_itemL1-bg-default;
+                    & {
+                        background-color: $color-navigation_itemL1-bg-default;
+                    }
```
Comment (pointhalo) [inline packages/semi-foundation/navigation/navigation.scss:443]: 这里加 & {} 的必要性是因为什么？

File: packages/semi-foundation/dropdown/dropdown.scss [hunk 2]
```diff
@@ -33,11 +33,12 @@ $module: #{$prefix}-dropdown;
         padding-bottom: $spacing-dropdown_title-paddingBottom;
         padding-left: $spacing-dropdown_title-paddingLeft;
         padding-right: $spacing-dropdown_title-paddingRight;
+        @include font-size-small;
+        cursor: default;
+
```
Comment (pointhalo) [inline packages/semi-foundation/dropdown/dropdown.scss:38]: 文档里没看到 关于 @include 的顺序说明。它跟 cursor 有先后顺序要求吗？

File: packages/semi-foundation/input/input.scss [hunk 3]
```diff
@@ -562,14 +562,13 @@ $module: #{$prefix}-input;
 
         background-color: $color-input_default-bg-default;
         color: $color-input_prefix-text-default;
+        @include font-size-regular;
+        flex-shrink: 0;
```
Comment (pointhalo) [inline packages/semi-foundation/input/input.scss:567]: 这里也是，flex-shrink是否可以排在 include 前面？

File: packages/semi-foundation/dropdown/dropdown.scss (hunk 2 above)
Comment (YyumeiZhang) [reply packages/semi-foundation/dropdown/dropdown.scss:38]: 本身没有顺序说明，取决于 mixin 中是否有嵌套的子项目，font-size-small 中无嵌套子项目，因此无需考虑顺序

File: packages/semi-foundation/input/input.scss (hunk 3 above)
Comment (YyumeiZhang) [reply packages/semi-foundation/input/input.scss:567]: 同上，可以，但是排在前后都不影响，因为 font-size-regular的 mixin 中无嵌套的子选择器

File: packages/semi-foundation/navigation/navigation.scss (hunk 1 above)
Comment (YyumeiZhang) [reply packages/semi-foundation/navigation/navigation.scss:443]: image

**Pull request merged**